src/xdg-desktop-portal.c

```diff
@@ -187,6 +188,33 @@ export_portal_implementation (GDBusConnection *connection,
   g_debug ("providing portal %s", g_dbus_interface_skeleton_get_info (skeleton)->name);
 }
 
+static void
+export_host_portal_implementation (GDBusConnection        *connection,
```

Comment:
It's a bit unfortunate that this function doesn't by itself make sure that only host apps can call interfaces exported by it. The burden is currently completely on the implementation.